packages/wrapped-keys/src/lib/api/export-private-key.ts

```diff
@@ -0,0 +1,55 @@
+import { decryptToString } from '@lit-protocol/encryption';
+
+import { CHAIN_ETHEREUM, LIT_PREFIX } from '../constants';
+import { fetchPrivateKeyMetadata } from '../service-client';
+import { ExportPrivateKeyParams, ExportPrivateKeyResult } from '../types';
+import {
+  getFirstSessionSig,
+  getPkpAccessControlCondition,
+  getPkpAddressFromSessionSig,
+} from '../utils';
+
+/** Exports a previously persisted private key from the wrapped keys service for direct use by the caller, along with the keys metadata
+ *
+ * @param { ExportPrivateKeyParams } params Parameters required to export the private key
+ */
+export async function exportPrivateKey(
+  params: ExportPrivateKeyParams
+): Promise<ExportPrivateKeyResult> {
+  const { pkpSessionSigs, litNodeClient } = params;
+
+  const sessionSig = getFirstSessionSig(pkpSessionSigs);
+  const pkpAddress = getPkpAddressFromSessionSig(sessionSig);
+  const allowPkpAddressToDecrypt = getPkpAccessControlCondition(pkpAddress);
+
+  const privateKeyMetadata = await fetchPrivateKeyMetadata({
+    sessionSig,
+    litNetwork: litNodeClient.config.litNetwork,
+  });
+
+  const { ciphertext, dataToEncryptHash, ...privateKeyMetadataMinusEncrypted } =
+    privateKeyMetadata;
+
+  const decryptedPrivateKey = await decryptToString(
+    {
+      accessControlConditions: [allowPkpAddressToDecrypt],
+      chain: CHAIN_ETHEREUM, // FIXME: This won't always be true.
+      ciphertext,
+      dataToEncryptHash,
+      sessionSigs: pkpSessionSigs,
+    },
+    litNodeClient
+  );
+
+  // It will be of the form lit_<privateKey>
+  if (!decryptedPrivateKey.startsWith(LIT_PREFIX)) {
+    throw new Error(
+      `PKey was not encrypted with salt; all wrapped keys must be prefixed with '${LIT_PREFIX}'`
```

Comment:
How about for Custom Wrapped Keys. They may choose not to follow our convention so this results in them encrypting their key and not being able to access it

Reply:
So, we store the encrypted data at rest, but we can't actually see the data when the client posts it to know if it's salted.  However, we definitely don't want someone to be storing un-salted keys on our backend -- this seems like the _only place_ we could reasonably tell them that they have stored something that they shouldn't have on the backend.  If they want to _dangerously store un-salted keys on our backend_, IMO they should write their own _dangerously written `export-private-key` LIT action too_. They are already bypassing our SDK to get data into the backend without a salt on it, so there is no expectation that our 'stock' SDK logic will work at that point.  The LIT SDK internals should be opinionated about security IMO. What do you think?